app/Actions/Import/FromServer.php

```diff
@@ -25,10 +25,12 @@ public function do(string $path, ?Album $album, ImportMode $importMode): Streame
 		$response->headers->set('Cache-Control', 'no-store');
 		// nginx-specific voodoo, as per https://symfony.com/doc/current/components/http_foundation.html#streaming-a-response
 		$response->headers->set('X-Accel-Buffering', 'no');
-		$response->setCallback(function () use ($path, $album, $exec) {
+		$response->setCallback(function () use ($paths, $album, $exec) {
 			// Surround the response by `[]` to make it a valid JSON array.
 			echo '[';
-			$exec->do($path, $album);
+			foreach ($paths as $path) {
+				$exec->do($path, $album);
+			}
 			echo ']';
```

Comment:
Because it is a streamed response. For a streamed response STDOUT is redirected to the body of the HTTP response. The `echo` and the `$exec->do` are part of an anonymous function which generates the steam. The brackets `[` and `]` are the beginning and end of a JSON array. `$exec->do` prints output as well and creates the entries within the JSON array.

Somewhere else (I believe it is in the action handler for uploads) it is a bit more explicit. There we have a static function instead of an anonymous function which is called `streamGenerator`. Maybe we should do that here as well.